presto-hive/src/main/java/io/prestosql/plugin/hive/HiveUpdatablePageSource.java

```diff
@@ -114,36 +131,58 @@ public HiveUpdatablePageSource(
         this.hiveRowTypeNullsBlock = nativeValueToBlock(hiveRowType.getType(typeManager), null);
         checkArgument(hiveTableHandle.isInAcidTransaction(), "Not in a transaction; hiveTableHandle: %s", hiveTableHandle);
         this.writeId = hiveTableHandle.getWriteId();
+        this.updateKind = requireNonNull(updateKind, "updateKind is null");
         this.hiveAcidSchema = createAcidSchema(hiveRowType);
         requireNonNull(bucketPath, "bucketPath is null");
+        Matcher matcher;
         if (originalFile) {
-            Matcher matcher = ORIGINAL_FILE_PATH_MATCHER.matcher(bucketPath.toString());
+            matcher = ORIGINAL_FILE_PATH_MATCHER.matcher(bucketPath.toString());
             checkArgument(matcher.matches(), "Original file bucketPath doesn't have the required format: %s", bucketPath);
             this.bucketFilename = format("bucket_%05d", bucketNumber.isEmpty() ? 0 : bucketNumber.getAsInt());
-            this.deleteDeltaDirectory = new Path(format("%s/%s", matcher.group("rootDir"), deleteDeltaSubdir(writeId, writeId, statementId)));
         }
         else {
-            Matcher matcher = BUCKET_PATH_MATCHER.matcher(bucketPath.toString());
-            checkArgument(matcher.matches(), "bucketPath doesn't have the required format: %s", bucketPath);
-            // delete_delta bucket files should not have attemptId suffix
-            this.bucketFilename = matcher.group("filenameBase");
-            this.deleteDeltaDirectory = new Path(format("%s/%s", matcher.group("rootDir"), deleteDeltaSubdir(writeId, writeId, statementId)));
+            matcher = BASE_PATH_MATCHER.matcher(bucketPath.toString());
+            if (matcher.matches()) {
+                this.bucketFilename = matcher.group("filenameBase");
+            }
+            else {
+                matcher = BUCKET_PATH_MATCHER.matcher(bucketPath.toString());
+                checkArgument(matcher.matches(), "bucketPath doesn't have the required format: %s", bucketPath);
+                this.bucketFilename = matcher.group("filenameBase");
```

Comment:
The base file and delta files can coexist as well before the subsequent compaction. Is that handled?

Reply:
It's a good point.  I just updated product test TestHiveTransactionalTable.testAcidUpdateMajorCompaction to add a series of operations that create more delta files in addition to the compacted base image, showing that they can coexist.